doc/musig-spec.mediawiki

```diff
@@ -60,7 +62,120 @@ The following conventions are used, with constants as defined for [https://www.s
+* The tweak ''t'': an integer with ''0 &le; t < n ''
+
+The algorithm ''OrdinaryTweak(P, t)'' is defined as:
+* Return P + t⋅G
+
+The algorithm ''XonlyTweak(P, t)'' is defined as:
+* Return set_even_y(P) + t⋅G
+
+=== Negation Of The Secret Key When Signing ===
+
+In order to produce a partial signature for an x-only public key that is an aggregate of x-only keys and x-only or ordinarily tweaked, the ''[[#Sign negation|Sign]]'' algorithm may need to negate the secret key during the signing process.
+
+<poem>
+The following public keys arise as intermediate steps in the MuSig protocol:
+• ''P<sub>i</sub>'' as computed in ''KeyAggInternal'' is the point corresponding to the ''i''-th signer's x-only public key. Defining ''d'<sub>i</sub>'' to be the ''d' '' value as computed in the ''Sign'' algorithm of the ''i''-th signer, we have
+    ''P<sub>i</sub> = set_even_y(d'<sub>i</sub>⋅G) ''.
+• ''Q<sub>0</sub>'' as defined in ''Tweak'' is the aggregate public key computed in ''KeyAggInternal'' as
+    ''Q<sub>0</sub> = a<sub>1</sub>⋅P<sub>1</sub> + a<sub>2</sub>⋅P<sub>1</sub> + ... + a<sub>u</sub>⋅P<sub>u</sub>''.
+• ''Q<sub>i</sub>'' as computed in ''Tweak'' for ''1 &le; i &le; v'' is the tweaked public key after the ''i''-th tweaking operation. It holds that
+    ''Q<sub>i</sub> = f(i-1) + t<sub>i</sub>⋅G'' for ''i = 1, ..., v'' where
+        ''f(i) := set_even_y(Q<sub>i</sub>)'' if ''is_xonly_t<sub>i+1</sub>'' and
+        ''f(i) := Q<sub>i</sub>'' otherwise.
+</poem>
+
+The goal is to produce a partial signature corresponding to the output of ''KeyAgg'', i.e., the final (x-only) public key point after ''v'' tweaking operations ''set_even_y(Q<sub>u</sub>)''.
+
+<poem>
+We define ''gp<sub>i</sub>'' for ''1 &le; i &le; u'' to be ''gp '' as computed in the ''Sign'' algorithm of the ''i''-th signer. It holds that
+    ''P<sub>i</sub> = gp<sub>i</sub>⋅d'<sub>i</sub>⋅G''.
+
+For ''0 &le; i &le; v-1'', the ''Tweak'' algorithm sets ''g<sub>i</sub>'' to ''-1'' if and only if ''is_xonly_t<sub>i+1</sub>'' is true and ''Q<sub>i</sub>'' has an odd Y coordinate. Therefore, we have
+    ''f(i) = g<sub>i</sub>⋅Q<sub>i</sub>'' for ''0 &le; i &le; v - 1''.
+
+Furthermore, the ''Tweak'' algorithm sets ''g<sub>v</sub>'' such that
+    ''set_even_y(Q<sub>v</sub>) = g<sub>v</sub>⋅Q<sub>v</sub>''.
+</poem>
+
+<poem>
+The (x-only) final public key is
+    ''set_even_y(Q<sub>v</sub>)
+        = g<sub>v</sub>⋅Q<sub>v</sub>
+        = g<sub>v</sub>⋅(f(v-1) + t<sub>v</sub>⋅G)
+        = g<sub>v</sub>⋅(g<sub>v-1</sub>⋅(f(v-2) + t<sub>v-1</sub>⋅G) + t<sub>v</sub>⋅G)
+        = g<sub>v</sub>⋅...⋅g<sub>0</sub>⋅Q<sub>0</sub> + (g<sub>v</sub>⋅t<sub>v</sub>+g<sub>v</sub>⋅g<sub>v-1</sub>⋅t<sub>v-1</sub> + ... + g<sub>v</sub>⋅g<sub>v-1</sub>⋅...⋅g<sub>1</sub>⋅t<sub>1</sub>)⋅G''.
+</poem>
+
+<poem>
+The ''KeyAgg'' algorithm computes
+    ''g = g<sub>v</sub>⋅...⋅g<sub>0</sub>
+    t = sum<sub>i=1</sub><sup>v</sup> t<sub>i</sub>⋅(prod<sub>j=i</sub><sup>v</sup> g<sub>j</sub> ) mod n
```

Comment:
sub and sup don't look nice together. I'd suggest to switch to a `sum<sub>i=1..v</sub>` style